File: capacity-club-api/src/domain-modules/member-home-training/model/payload/member-home-training-create.payload.ts
import { ApiProperty } from '@nestjs/swagger';
import { HomeTraining } from 'domain-modules/home-training/model/entity/home-training.entity';
import { Member } from 'domain-modules/member/model/entity/member.entity';

/**
 * Class used to create member home training.
 * This class defines the payload used when creating new member home training.
 */
export class MemberHomeTrainingCreatePayload {
  @ApiProperty()
  purchase_date: Date;
  @ApiProperty({ type: () => Member })
  member: Member;
  @ApiProperty({ type: () => HomeTraining })
  home_training: HomeTraining;
}

File: capacity_club_mobile_app/lib/domain-features/data/member-home-training/model/payload/member_home_training_create_payload.dart
import 'package:capacity_club_mobile_app/domain-features/data/home-training/model/home_training_model.dart';
import 'package:capacity_club_mobile_app/domain-features/data/member/model/member_model.dart';
import 'package:equatable/equatable.dart';

class MemberHomeTrainingCreatePayload extends Equatable {
  final DateTime purchase_date;
  final HomeTrainingModel home_training;
  final MemberModel member;

  const MemberHomeTrainingCreatePayload({
    required this.purchase_date,
    required this.home_training,
    required this.member,
  });

  Map<String, dynamic> toJson() => {
        'purchase_date': purchase_date.toIso8601String(),
        'home_training': home_training.toJson(),
        'member': member.toJson(),
      };

  @override
  List<Object?> get props => [
        purchase_date,
        home_training,
        member,
      ];
}
